put-object.go

```diff
@@ -180,26 +189,19 @@ func (a API) putNoChecksum(bucketName, objectName string, data io.ReadSeeker, si
 		ContentType: contentType,
 	}
 	if _, err := a.putObject(bucketName, objectName, putObjMetadata); err != nil {
-		return err
+		return 0, err
 	}
-	return nil
+	return size, nil
 }
 
 // putSmallObject uploads files smaller than 5 mega bytes.
-func (a API) putSmallObject(bucketName, objectName string, data io.ReadSeeker, size int64, contentType string) error {
+func (a API) putSmallObject(bucketName, objectName string, data io.ReadSeeker, size int64, contentType string) (int64, error) {
 	dataBytes, err := ioutil.ReadAll(data)
 	if err != nil {
-		return err
+		return 0, err
 	}
 	if int64(len(dataBytes)) != size {
-		msg := fmt.Sprintf("Data read ‘%s’ is not equal to expected size ‘%s’",
-			strconv.FormatInt(int64(len(dataBytes)), 10), strconv.FormatInt(size, 10))
-		return ErrorResponse{
-			Code:       "UnexpectedShortRead",
-			Message:    msg,
-			BucketName: bucketName,
-			Key:        objectName,
-		}
+		return int64(len(dataBytes)), ErrUnexpectedEOF(int64(len(dataBytes)), size, bucketName, objectName)
```

Comment:
return int64(len(dataBytes)), ErrUnexpectedEOF(int64(len(dataBytes)), size, bucketName, objectName)
should be
return 0, ErrUnexpectedEOF(int64(len(dataBytes)), size, bucketName, objectName)

Because this size is reserved for PutObject's written len use.